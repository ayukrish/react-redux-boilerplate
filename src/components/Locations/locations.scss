.locationWrapper {
  justify-content: space-around;
  padding: 20px;
}