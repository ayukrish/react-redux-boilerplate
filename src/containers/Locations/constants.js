export const GET_LOCATIONS = 'GET_LOCATIONS';
export const SET_LOCATIONS = 'SET_LOCATIONS';
