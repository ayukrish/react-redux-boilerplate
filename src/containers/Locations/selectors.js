export const locationData = state => state.location && state.location.data;
