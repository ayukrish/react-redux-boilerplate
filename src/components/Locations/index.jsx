import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Card from '../Card';
import Pagination from '../Pagination';
import * as styles from './locations.scss';

const propTypes = {
  getLocations: PropTypes.func,
  locationData: PropTypes.object
};

const defaultProps = {
  locationData: {},
  getLocations: () => {}
};

export default class Locations extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      currentPage: 1
    };
  }

  componentDidMount() {
    this.props.getLocations(this.state.currentPage);
  }

  render() {
    const { locationData } = this.props;
    const { results = {} } = locationData;
    return (
      <section className={`${styles.locationWrapper} flex wrap`}>
        {results.length > 0 &&
          results.map((item) => (
            <Card
              key={item.id}
              dataObj={{
                TYPE: item && item.type,
                DIMENSION: item && item.dimension
              }}
              heading={item && item.name}
            />
          ))}
        <Pagination
          contentLength={(results && results.length) || 0}
          currentPage={this.state.currentPage}
          limit={20}
          onChange={(currentPage) => {
            this.props.getLocations(currentPage);
            this.setState({
              currentPage
            });
          }}
        />
      </section>
    );
  }
}

Locations.propTypes = propTypes;
Locations.defaultProps = defaultProps;
