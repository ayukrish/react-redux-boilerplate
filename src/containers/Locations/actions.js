import { GET_LOCATIONS, SET_LOCATIONS } from './constants';


export const getLocations = (pageNo) => ({ type: GET_LOCATIONS, data: pageNo });

export const setLocations = data => ({ type: SET_LOCATIONS, data });
