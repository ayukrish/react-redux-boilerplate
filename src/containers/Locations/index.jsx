import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { getLocations } from './actions';
import Locations from '../../components/Locations';
import { locationData } from './selectors';

const mapStateToProps = createStructuredSelector({
  locationData
});

const mapDispatchToProps = (dispatch) => ({
  getLocations: (pageNo) => dispatch(getLocations(pageNo))
});

export default connect(mapStateToProps, mapDispatchToProps)(Locations);
