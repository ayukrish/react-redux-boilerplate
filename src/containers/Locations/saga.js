import { takeLatest, put, call } from 'redux-saga/effects';
import { service } from '../../service';
import { enpoints } from '../../endpoint';
import { setLocations } from './actions';
import { GET_LOCATIONS } from './constants';

export function* getLocationsSaga({ data }) {
  try {
    // data is obtained after axios call is resolved
    const response = yield call(service, {
      url: enpoints.locations,
      method: 'get',
      page: data
    });
    yield put(setLocations(response));
  } catch (e) {
    // catch error on a bad axios call
    // alert using an alert library
  }
}

export function* getLocationSaga() {
  yield takeLatest(GET_LOCATIONS, getLocationsSaga);
}
