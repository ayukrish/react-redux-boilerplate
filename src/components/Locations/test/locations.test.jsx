import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

// Components
import Locations from '../index';

describe('Locations Test Suite', () => {
  it('renders correctly without any data', () => {
    const output = shallow(<Locations />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('renders correctly without any data', () => {
    const output = shallow(
      <Locations
        getLocations={() => {}}
        locationData={{
          results: [
            {
              id: 1,
              name: 'Earth (C-137)',
              type: 'Planet',
              dimension: 'Dimension C-137'
            }
          ]
        }}
      />
    );
    output.find('Pagination').props().onChange(1);
    expect(output.state().currentPage).toEqual(1);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
