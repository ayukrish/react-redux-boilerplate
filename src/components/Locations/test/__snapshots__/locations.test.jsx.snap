// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Locations Test Suite renders correctly without any data 1`] = `
<section
  className="undefined flex wrap"
>
  <Pagination
    contentLength={0}
    currentPage={1}
    limit={20}
    onChange={[Function]}
  />
</section>
`;

exports[`Locations Test Suite renders correctly without any data 2`] = `
<section
  className="undefined flex wrap"
>
  <Card
    dataObj={
      Object {
        "DIMENSION": "Dimension C-137",
        "TYPE": "Planet",
      }
    }
    heading="Earth (C-137)"
    imgSrc=""
    key="1"
  />
  <Pagination
    contentLength={1}
    currentPage={1}
    limit={20}
    onChange={[Function]}
  />
</section>
`;
